code/game/objects/structures/maintenance.dm

```diff
@@ -105,3 +105,70 @@ at the cost of risking a vicious bite.**/
 		return
 	hidden_item = I
 	to_chat(user, span_notice("You hide [I] inside the basin."))
+
+/**
+ * Spawns in maint shafts, and blocks lines of sight perodically when active.
+ */
+/obj/structure/steam_vent
+	name = "steam vent"
+	desc = "A device periodically filtering out moisture particles from the nearby walls and windows. It's only possible due to the moisture traps nearby."
+	icon_state = "steam_vent"
+	anchored = TRUE
+	density = FALSE
+	/// How often does the vent reset the blow_steam callback loop.
+	var/steam_speed = 15 SECONDS
+	/// Is the steam vent active?
+	var/vent_active = TRUE
+	///The cooldown for toggling the steam vent to prevent infinite steam vent looping.
+	COOLDOWN_DECLARE(steam_vent_interact)
+
+/obj/structure/steam_vent/Initialize(mapload)
+	. = ..()
+	if(prob(75))
+		vent_active = FALSE
+	var/timer = (steam_speed - (rand(1,5) SECONDS))
+	addtimer(CALLBACK(src, .proc/blow_steam), timer)
```

Comment:
(Old maintainer :older_man: , so maybe things have changed over the years)
I thought using timers to schedule additional timers created a lot of strain on the Timer subsystem, and it was better to just use Processing instead for these sorts of things.

Reply:
Yeah, this I wasn't sure of that myself, I considered just going and attaching it to a subsystem, but that would prevent it from being stagger-able.  If we have to move it onto a subsystem though I totally get it.